samples/bluetooth/central_gatt_write/src/gatt_write_common.c

```diff
@@ -0,0 +1,226 @@
+	err = bt_gatt_exchange_mtu(conn, &mtu_exchange_params);
+	if (err) {
+		printk("%s: MTU exchange failed (err %d)", __func__, err);
+	}
+
+	return err;
+}
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	struct bt_conn_info conn_info;
+	char addr[BT_ADDR_LE_STR_LEN];
+	int err;
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		printk("%s: Failed to connect to %s (%u)\n", __func__, addr,
+		       conn_err);
+		return;
+	}
+
+	err = bt_conn_get_info(conn, &conn_info);
+	if (err) {
+		printk("Failed to get connection info (%d).\n", err);
+		return;
+	}
+
+	printk("%s: %s role %u\n", __func__, addr, conn_info.role);
+
+	conn_connected = bt_conn_ref(conn);
+
+	(void)mtu_exchange(conn);
+
+#if defined(CONFIG_BT_SMP)
+	if (conn_info.role == BT_CONN_ROLE_CENTRAL) {
+		err = bt_conn_set_security(conn, BT_SECURITY_L2);
+		if (err) {
+			printk("Failed to set security (%d).\n", err);
+		}
+	}
+#endif
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	struct bt_conn_info conn_info;
+	char addr[BT_ADDR_LE_STR_LEN];
+	int err;
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	err = bt_conn_get_info(conn, &conn_info);
+	if (err) {
+		printk("Failed to get connection info (%d).\n", err);
+		return;
+	}
+
+	printk("%s: %s role %u (reason %u)\n", __func__, addr, conn_info.role,
+	       reason);
+
+	conn_connected = NULL;
+
+	bt_conn_unref(conn);
+
+	if (conn_info.role == BT_CONN_ROLE_CENTRAL) {
+		start_scan_func();
+	}
+}
+
+static bool le_param_req(struct bt_conn *conn, struct bt_le_conn_param *param)
+{
+	printk("%s: int (0x%04x, 0x%04x) lat %u to %u\n", __func__,
+	       param->interval_min, param->interval_max, param->latency,
+	       param->timeout);
+
+	return true;
+}
+
+static void le_param_updated(struct bt_conn *conn, uint16_t interval,
+			     uint16_t latency, uint16_t timeout)
+{
+	printk("%s: int 0x%04x lat %u to %u\n", __func__, interval,
+	       latency, timeout);
+}
+
+#if defined(CONFIG_BT_SMP)
+static void security_changed(struct bt_conn *conn, bt_security_t level,
+			     enum bt_security_err err)
+{
+	printk("%s: to level %u (err %u)\n", __func__, level, err);
+}
+#endif
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+	.le_param_req = le_param_req,
+	.le_param_updated = le_param_updated,
+#if defined(CONFIG_BT_SMP)
+	.security_changed = security_changed,
+#endif
+};
+
+int write_cmd(struct bt_conn *conn)
+{
+	static uint8_t data[BT_ATT_MAX_ATTRIBUTE_LEN] = {0, };
+	static uint16_t data_len;
+	uint16_t data_len_max;
+	int err;
+
+	data_len_max = bt_gatt_get_mtu(conn) - 3;
+	if (data_len_max > BT_ATT_MAX_ATTRIBUTE_LEN) {
+		data_len_max = BT_ATT_MAX_ATTRIBUTE_LEN;
+	}
+
+#if TEST_FRAGMENTATION_WITH_VARIABLE_LENGTH_DATA
+	/* Use incremental length data for every write command */
+	/* TODO: Include test case in BabbleSim tests */
+	static uint8_t direction;
```

Comment:
I am still unsure about this value. What does `direction` reference to? 

It looks like it is being used to either increment or decrement the `data_len`. It is also being used like a boolean value, but is an `uint8_t`. 

Perhaps it should have been `bool decrement;` instead?